src/sdk/cpp/lib/src/protocol_gateway.cpp

```diff
@@ -164,12 +186,22 @@ void ProtocolGateway::Initialize() {
         }
 
         auto adapter = load(module_name, is_platform_proto, module_config);
-        adapters_.push_back(adapter);
+        Add(adapter);
```

Comment:
Improvement: Add error message if the return of Add(...) is false

Reply:
Will do.